botocore/configprovider.py

```diff
@@ -147,6 +147,20 @@
     # whatever the defaults are in _retry.json.
     'max_attempts': ('max_attempts', 'AWS_MAX_ATTEMPTS', None, int),
     'user_agent_appid': ('sdk_ua_app_id', 'AWS_SDK_UA_APP_ID', None, None),
+    # This must be a parsable integer between 0 and 1048576, but validation
+    # is performed during client initialization instead of here.
```

Comment:
We can leave this to the validation and comment in the client to avoid needing to update this in many places.